### PR TITLE
Use active filter in async_bulk

### PR DESCRIPTION
**Pull request by wangch079**
### Changes made:

Pass the active filtering (an instance of `Filter`) instead of filtering (a list of filtering including `active` and `draft` filtering, an instance of `Filtering`).

### Why the change:

Moving forward, we will use `filtering` data in `syncJob`, where only active filtering data is available.

#### Pre-Review Checklist
- [ ] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/byoei.py
Comment (tarekziade) [inline connectors/byoei.py:180]: if you use `Filter` instead of `Filtering`, shouldn't we rename the `filtering` argument as well?

Comment (wangch079) [reply connectors/byoei.py:180]: I know it's a bit confusing, we actually use `filtering` for everything interchangeably. 

This is the structure of the connector filtering:
```
"filtering": [
  {
    "domain": "DEFAULT",
    "draft": {"advanced_snippet": {}, "rules": [], "validation": {}},
    "active": {"advanced_snippet": {}, "rules": [], "validation": {}}
  }
]
```

The `Filtering` instance represents the outer array, but there should only be one element, which we filter by `domain`, and the `Filter` instance represents either the `draft` or `active` filtering. 

I think we should have some proper naming. cc @timgrein `Filterings` and `Filtering`? I don't think it's a good idea to name a class plural.

Comment (timgrein) [reply connectors/byoei.py:180]: Maybe we should go for `FilterCollection`, which contains multiple `Filter` instances. The assumption that we always only have one `Filter` instance inside `filtering` will probably not be true anymore in 8.8 as we may look into using separate `Filter` instances for different tables (as an example).

++ on class names shouldn't be plural, would make it harder to distinguish between `Filters` and `Filter`

https://medium.com/swlh/an-opinionated-guide-to-naming-your-code-aimed-at-new-developers-6f769d60e098 mentions some further reasons under "Class names"

**Comment by wangch079** [comment]
This PR includes some changes in PR #440 , which is being blocked.

Will only merge this after #440 is merged.

File: connectors/byoei.py
Comment (wangch079) [reply connectors/byoei.py:180]: I kind of think should rename `Filter` to `Filtering`, as this is the first and only place I see it's named like this.

In the Project brief, its official name is `Filtering Rules`. In UI, it's Sync Rules.

WDYT if we rename `Filtering` to `FilteringSetting`, and rename `Filter` to `Filtering`.

Speaking of plural class name, we do have class `PipelineSettings`.

Comment (timgrein) [reply connectors/byoei.py:180]: My personal preference would be to have a class name, which makes it obvious that we're having multiple instances of a specific type at hand. A `FilteringSetting` could also be _one_ filtering setting for _one_ domain. From the workplace search docs I see that they're using the term _filters_. Maybe @danajuratoni can help, how we'll call a single filter/filtering/filtering setting (or whatever) in the future and base our decision on that?

But for me it's fine as long as it's consistent and obvious what's the container and what's the single instance. I don't want to block here 👍

Comment (wangch079) [reply connectors/byoei.py:180]: Agree, I don't think this is blocking. Not customer-facing. I will just rename this as `Filter`.

**Review by artem-shelkovnikov — APPROVED** [review]
I think it's okay, but I'm a little away from filter/filtering design, @timgrein how do you feel?

**Pull request merged**
